pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -224,3 +228,27 @@ func countEnabledFields(fields ...*string) int {
 	}
 	return count
 }
+
+func validateNodeGroupKubeletExtraConfig(kubeletConfig *NodeGroupKubeletConfig) error {
+	if kubeletConfig == nil {
+		return nil
+	}
+
+	var kubeletForbiddenFields = map[string]struct{}{
+		"kind":               {},
+		"apiVersion":         {},
+		"address":            {},
+		"clusterDomain":      {},
+		"authentication":     {},
+		"authorization":      {},
+		"serverTLSBootstrap": {},
+		"featureGates":       {},
+	}
+
+	for k := range *kubeletConfig {
+		if _, exists := kubeletForbiddenFields[k]; exists {
+			return fmt.Errorf("kubelet key %q cannot be overwritten", k)
```

Comment:
how about `"cannot override %q in kubelete config, as it's critical to eksctl functionality"`?